app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -504,4 +313,69 @@ public void onServiceDisconnected(ComponentName arg0) {
             mBound = false;
         }
     };
+
+    private String findSmsData(String[] prebuffer,String[] postbuffer){
```

Comment:
No this is efficient the loops aren't used if null is passed, if you have a way of checking 2 arrays in same loop than Im all for it.